nmma/em/analysis.py

```diff
@@ -540,16 +540,27 @@ def analysis(args):
         # load the kilonova afterglow data
         try:
             data = loadEvent(args.data)
+            
+            #load the minimum time to use if args.trigger is None
+            #TODO: this code will be moved below, once the structue of json is identified
+            
+            min_time = np.inf
+            for key, array in data.items():
+                min_time = np.minimum(min_time, np.min(array[:, 0]))
+
         except ValueError:
             with open(args.data) as f:
                 data = json.load(f)
                 for key in data.keys():
                     data[key] = np.array(data[key])
 
         if args.trigger_time is None:
-            raise ValueError("trigger_time required if using a data file.")
-
-        trigger_time = args.trigger_time
+            print("trigger_time is not provided, analysis will continue using the time of first data point") #maybe change it to warning instead of print
+            trigger_time=min_time #For now this will only work if data file is not in JSON or try block is executed successfully.
```

Comment:
We should probably support json lightcurves, at least in the format that nmma itself creates from injections

Reply:
But yeah, this seems to be in line with what I was thinking for handling when a trigger time wasn't provided

Reply:
@tylerbarna Do you have any json LC for test?

Reply:
> @tylerbarna Do you have any json LC for test?

@sahiljhawar [here's one](https://github.com/tylerbarna/dsmma_kn_23/blob/model-recovery/injections/lc_Me2017_00000.json)